casa/IO/test/tByteSinkSource.cc

```diff
@@ -69,10 +67,10 @@ int main ()
 	uChar    testuChar = 'B';
 	String   testString("This is a teststring");
 
-	RegularFileIO regularFileIO (Path("tByteSinkSource_tmp.dat1"),
-				     ByteIO::New);
-	CanonicalIO canonicalIO(&regularFileIO);
-	ByteSinkSource  sinkSource(&canonicalIO);
+        std::shared_ptr<ByteIO> regularFileIO
+          (new RegularFileIO (Path("tByteSinkSource_tmp.dat1"), ByteIO::New));
+        std::shared_ptr<TypeIO> canonicalIO(new CanonicalIO(regularFileIO));
```

Comment:
```suggestion
        std::shared_ptr<RegularFileIO> regularFileIO
          (new RegularFileIO (Path("tByteSinkSource_tmp.dat1"), ByteIO::New));
        std::shared_ptr<CanonicalIO> canonicalIO(new CanonicalIO(regularFileIO));
```
Additionally, you could use `std::make_shared` here.